src/app/loadout-builder/loadout-params.ts

```diff
@@ -0,0 +1,107 @@
+/* Functions for dealing with the LoadoutParameters structure we save with loadouts and use to save and share LO settings. */
+
+import {
+  LoadoutParameters,
+  StatConstraint,
+  UpgradeSpendTier,
+} from '@destinyitemmanager/dim-api-types';
+import { D2ManifestDefinitions } from 'app/destiny2/d2-definitions';
+import { PluggableInventoryItemDefinition } from 'app/inventory/item-types';
+import { isInsertableArmor2Mod } from 'app/loadout/mod-utils';
+import { armorStats } from 'app/search/d2-known-values';
+import { Settings } from 'app/settings/initial-settings';
+import _ from 'lodash';
+import { ArmorStatHashes, MinMaxIgnored, StatFilters } from './types';
+
+export function buildLoadoutParams(
+  upgradeSpendTier: UpgradeSpendTier,
+  lockItemEnergyType: boolean,
+  lockedMods: PluggableInventoryItemDefinition[],
+  searchQuery: string,
+  statFilters: Readonly<{ [statType in ArmorStatHashes]: MinMaxIgnored }>,
+  statOrder: number[],
+  exoticArmorHash?: number
+): LoadoutParameters {
+  const params: LoadoutParameters = {
+    statConstraints: _.compact(
+      statOrder.map((statHash) => {
+        const minMax = statFilters[statHash];
+        if (minMax.ignored) {
+          return undefined;
+        }
+        const stat: StatConstraint = {
+          statHash,
+        };
+        if (minMax.min > 0) {
+          stat.minTier = minMax.min;
+        }
+        if (minMax.max < 10) {
+          stat.maxTier = minMax.max;
+        }
+        return stat;
+      })
+    ),
+    lockItemEnergyType,
+    upgradeSpendTier,
+  };
+
+  if (lockedMods) {
+    params.mods = lockedMods.map((mod) => mod.hash);
+  }
+  if (searchQuery) {
+    params.query = searchQuery;
+  }
+  if (exoticArmorHash) {
+    params.exoticArmorHash = exoticArmorHash;
+  }
+
+  return params;
+}
+
+/**
+ * Given a Settings object, migrate the old-style toplevel LO settings into the new LoadoutParameters setting.
+ */
+export function migrateLoadoutParametersFromSettings(settings: Settings): Settings {
+  // Only migrate if there aren't already settings
+  if (_.isEmpty(settings.loParameters)) {
+    return {
+      ...settings,
+      loParameters: {
+        statConstraints: settings.loStatSortOrder.map((statHash) => ({
+          statHash,
+        })),
+        upgradeSpendTier: settings.loUpgradeSpendTier,
```

Comment:
Remember when I removed that enum value and have had to put the guard in, in the event that someone had that version somehow. That could potentially be migrated to here.

Reply:
Got it. I've updated it and we can remove it from the code after a bit

Reply:
Sounds great